### ISSUE TITLE
Safari issue

### ISSUE THREAD

**Issue by atanasster** [tensorflow/tfjs#287]
Hi, I get the following issue in Safari, same site works fine in Chrome and Firefox:

`Unhandled Promise Rejection: Error: Dropout is not implemented for LSTMCell yet`
I get it right after the known Safari issue `WebGL: INVALID_ENUM: readPixels: invalid type`

Let me know if you need the deployment site and code

**Comment by atanasster** [comment]
with tf.setBackend('cpu'); works fine in Safari, seems it WebGL related

**Comment by nsthorat** [comment]
Unfortunately "WebGL: INVALID_ENUM: readPixels: invalid type" comes from when we feature test readPixels on Safari and is a red herring.

Does your application still work?

**Comment by atanasster** [comment]
No, the app dissent work. I had to start detecting Safari browser and drop to CPU if detected. 
I understand about INVALID_ENUM, but the `Unhandled Promise Rejection: Error: Dropout is not implemented for LSTMCell yet` is the one stopping the app. 
Dense layers work fine on GPU even with the `INVALID_ENUM` error

**Comment by davidsoergel** [comment]
Can you call getConfig() on your LSTM layer to see what the values of the dropout and recurrentDropout fields are?  Since you're not passing them in, the constructor logic should set them to 0.  But: that error indicates that at least one of them is something else, and it would be informative to know what.

**Comment by atanasster** [comment]
I have this: 
```
"dropout": 0,
"recurrentDropout": 0,
```
both in 5.6 and 6

**Comment by atanasster** [comment]
Are you able to reproduce, do you need drivers and hardware specs?

**Comment by davidsoergel** [comment]
Yep I repro'd this via your app link in #288 but don't understand it yet.

To help narrow down the cause: can you try explicitly passing dropout=0 and recurrentDropout=0L

**Comment by atanasster** [comment]
I tried, and it still happens. Actually you can use the editor (click on the gears next to the LSTM layer) and change most any values if that might help to understand whats going on.
I also have the code in the GitHub link, might this help? 
Just let me know if you need any of my system information, and also when I can put back my Safari check to drop down to CPU
